perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathChangeListener.java

```diff
@@ -0,0 +1,18 @@
+package name.mitterdorfer.perlock;
+
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+
+/**
+ * Callback interface for notifying the user when the chosen file changes.
+ * Suited for use as a functional interface
+ *
+ * <p>Implementation note: A <code>PathChangeListener</code> will be called from the internal <code>PathWatcher</code>
+ * thread. This implies that callback methods should return reasonably fast and offload heavy lifting to a dedicated
+ * thread. Otherwise, events may be lost.</p>
+ */
+public interface SinglePathChangeListener {
```

Comment:
I think this interface is great and better than the existing `PathChangeListener` (especially since it can be used as a functional interface in Java 8+). Would you be willing you replace `PathChangeListener` with your interface in general in Perlock? I think it is fine as Perlock is at the moment in version 0.3 and we can introduce breaking API changes.

This would also eliminate the need for `AbstractPathChangeListener`.